File: content-solutions/greedy/balloon-arrows.java
import java.util.*;

class Solution {
    public int solution(int[][] balloons) {
        int[][] sorted = balloons.clone();
        Arrays.sort(sorted, (a, b) -> Integer.compare(a[1], b[1]));
        long arrow = Long.MIN_VALUE;
        int count = 0;
        for (int[] b : sorted) {
            if (b[0] > arrow) {
                count++;
                arrow = b[1];
            }
        }
        return count;
    }
}

File: content-solutions/greedy/balloon-arrows.js
function solution(balloons) {
  let arrow = -Infinity;
  let count = 0;
  for (const [l, r] of [...balloons].sort((a, b) => a[1] - b[1])) {
    if (l > arrow) {
      count++;
      arrow = r;
    }
  }
  return count;
}
